baseplate/server/__init__.py

```diff
@@ -47,9 +50,19 @@
 logger = logging.getLogger(__name__)
 
 
+class ServerLifecycle(Enum):
+    RUNNING = enum.auto()
+    SHUTTING_DOWN = enum.auto()
```

Comment:
This is great!  Is the ordering sensitive?  (It almost certainly doesn't matter unless you're serializing stuff, which is hard to imagine)

Reply:
yeah, it's ordering sensitive. if we added a new one inbetween it'd mess up the latter but this state should only ever make sense in-process